### PR TITLE
`weather-mv` – Fixed error writing to BigQuery: Excluding non-coordinate indexes if they don't appear in the Schema

### LINKED ISSUE #94: `weather-mv`: Errors inserting into BigQuery during streaming uploads.
I just noticed in an e2e run, the pipeline doesn't fail but issues warnings about failure to insert data into BigQuery: 

```
There were errors inserting to BigQuery. Will retry. Errors were [{'index': 0, 'errors': [{'reason': 'invalid', 'location': 'step', 'debugInfo': '', 'message': 'no such field: step.'}]}, {'index': 1, 'errors': [{'reason': 'invalid', 'location': 'step', 'debugInfo': '', 'message': 'no such field: step.'}]}, {'index': 2, 'errors': [{'reason': 'invalid', 'location': 'surface', 'debugInfo': '', 'message': 'no such field: surface.'}]}, {'index': 3, 'errors': [{'reason': 'invalid', 'location': 'valid_time', 'debugInfo': '', 'message': 'no such field: valid_time.'}]}, {'index': 4, 'errors': [{'reason': 'invalid', 'location': 'surface', 'debugInfo': '', 'message': 'no such field: surface.'}]}, {'index': 5, 'errors': [{'reason': 'invalid', 'location': 'valid_time', 'debugInfo': '', 'message': 'no such field: valid_time.'}]}, {'index': 6, 'errors': [{'reason': 'invalid', 'location': 'surface', 'debugInfo': '', 'message': 'no such field: surface.'}]}, {'index': 7, 'errors': [{'reason': 'invalid', 'location': 'surface', 'debugInfo': '', 'message': 'no such field: surface.'}]}, {'index': 8, 'errors': [{'reason': 'invalid', 'location': 'surface', 'debugInfo': '', 'message': 'no such field: surface.'}]}, {'index': 9, 'errors': [{'reason': 'invalid', 'location': 'step', 'debugInfo': '', 'message': 'no such field: step.'}]}, {'index': 10, 'errors': [{'reason': 'invalid', 'location': 'number', 'debugInfo': '', 'message': 'no such field: number.'}]}, {'index': 11, 'errors': [{'reason': 'invalid', 'location': 'number', 'debugInfo': '', 'message': 'no such field: number.'}]}, {'index': 12, 'errors': [{'reason': 'invalid', 'location': 'valid_time', 'debugInfo': '', 'message': 'no such field: valid_time.'}]}, {'index': 13, 'errors': [{'reason': 'invalid', 'location': 'surface', 'debugInfo': '', 'message': 'no such field: surface.'}]}, {'index': 14, 'errors': [{'reason': 'invalid', 'location': 'valid_time', 'debugInfo': '', 'message': 'no such field: valid_time.'}]}, {'index': 15, 'errors': [{'reason': 'invalid', 'location': 'number', 'debugInfo': '', 'message': 'no such field: number.'}]}, {'index': 16, 'errors': [{'reason': 'invalid', 'location': 'step', 'debugInfo': '', 'message': 'no such field: step.'}]}, {'index': 17, 'errors': [{'reason': 'invalid', 'location': 'step', 'debugInfo': '', 'message': 'no such field: step.'}]}, {'index': 18, 'errors': [{'reason': 'invalid', 'location': 'step', 'debugInfo': '', 'message': 'no such field: step.'}]}, {'index': 19, 'errors': [{'reason': 'invalid', 'location': 'number', 'debugInfo': '', 'message': 'no such field: number.'}]}, {'index': 20, 'errors': [{'reason': 'invalid', 'location': 'step', 'debugInfo': '', 'message': 'no such field: step.'}]}, {'index': 21, 'errors': [{'reason': 'invalid', 'location': 'valid_time', 'debugInfo': '', 'message': 'no such field: valid_time.'}]}, {'index': 22, 'errors': [{'reason': 'invalid', 'location': 'step', 'debugInfo': '', 'message': 'no such field: step.'}]}, {'index': 23, 'errors': [{'reason': 'invalid', 'location': 'step', 'debugInfo': '', 'message': 'no such field: step.'}]}, {'index': 24, 'errors': [{'reason': 'invalid', 'location': 'valid_time', 'debugInfo': '', 'message': 'no such field: valid_time.'}]}]
```

I think the issue is this logic in `extract_rows`: https://github.com/google/weather-tools/blob/2d8651d212cc998f098cbcf83a6e4536e9e455a7/weather_mv/loader_pipeline/pipeline.py#L289 These lines don't match anything in the schema.

### PR DESCRIPTION
**Pull request by alxmrs**
I've added tests to confirm a corner case: When users supply variables for the BQ schema, we added extra columns in extract_rows that were not represented in the schema. This change addresses this in two ways:
- We've made it possible supply non-data_vars to the `-v` argument. This way, users can get non-indexed coordinate info like `step` and `valid_time` without inferring the schema.
- If `-v` is supplied, we skip adding non-explicit non-indexed coordinate info to the extracted rows.

This bug primarily affects streaming ingestion of weather-data into BQ. It causes a somewhat silent failure, as no data gets written since BQ enters an infinite retry loop (only during streaming ingestion).

Ideally, this should fix #94.

### CODE REVIEW

File: weather_mv/loader_pipeline/pipeline_test.py [hunk 1]
```diff
@@ -50,6 +56,10 @@ def setUp(self) -> None:
                 "d": {"dims": ("a",), "data": [3.0]},
             }
         }
+        self.test_single_var = xr.open_dataset(
```
Comment (pramodg) [inline weather_mv/loader_pipeline/pipeline_test.py:59]: This may not work on machines that do not have the grib backend setup.

File: weather_mv/README.md [hunk 2]
```diff
@@ -25,7 +25,8 @@ _Common options_:
 
 * `-i, --uris`: (required) URI prefix matching input netcdf objects. Ex: gs://ecmwf/era5/era5-2015-""
 * `-o, --output_table`: (required) Full name of destination BigQuery table. Ex: my_project.my_dataset.my_table
-* `-v, --variables`:  Target variables for the BigQuery schema. Default: will import all data variables as columns.
+* `-v, --variables`:  Target variables (or coordinates) for the BigQuery schema. Default: will import all data variables
```
Comment (pramodg) [inline weather_mv/README.md:28]: s/coordinates/non-indexed coordinates/ ?
(or a better word)

**Review by pramodg — APPROVED** [review]

File: weather_mv/README.md (hunk 2 above)
Comment (alxmrs) [reply weather_mv/README.md:28]: Yeah, I can see how leaving it this vague would be confusing. If users supply variables, we hard-code adding lat/lng and time. Further, the indexed-coordinates should already be added at the time we check for them in `to_rows()`.  So, in some sense, we allow adding both coordinates and non-indexed coordinates. 

Given all this, I think that coordinates in this text is accurate. WDYT?

File: weather_mv/loader_pipeline/pipeline_test.py (hunk 1 above)
Comment (alxmrs) [reply weather_mv/loader_pipeline/pipeline_test.py:59]: Fair point -- I'll use our Grib backend handler.
